### PR TITLE
Simple Fibrations and CT-Structures

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

Been reading Jacobs recently and thought these would be fun. This PR defines the simple fibration associated with a category with binary products, and also defines CT-structures and their associated simple fibration.

### CODE REVIEW

File: src/Cat/Displayed/Instances/CT-Structure.lagda.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  f                   ∎
+simple-ct ct .Displayed.assoc′ {f = u} {g = v} {h = w} f g h =
+  f ∘ ⟨ (v ∘ w) ∘ π₁ , g ∘ ⟨ w ∘ π₁ , h ⟩ ⟩ ≡⟨ products! B has-prods ⟩
+  (f ∘ ⟨ v ∘ π₁ , g ⟩) ∘ ⟨ w ∘ π₁ , h ⟩     ∎
+```
+
+# Fibration Structure
+
+Much like the simple fibration, the simple fibration associated to a
+CT-structure also deserves it's name.
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/CT-Structure.lagda.md:74]: its!

File: src/Cat/Displayed/Instances/CT-Structure.lagda.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+is that we restrict the displayed object to objects that the
+CT-structure distinguishes as types.
+
+[simple fibrations]: Cat.Displayed.Instances.Simple
+
+```agda
+simple-ct : ∀ {s} → CT-Structure s → Displayed B (o ⊔ s) ℓ
+Displayed.Ob[ simple-ct ct ] Γ = Σ[ X ∈ Ob ] ∣ is-tp ct X ∣
+Displayed.Hom[ simple-ct ct ] {Γ} {Δ} u X Y = Hom (Γ ⊗ X .fst) (Y .fst)
+Displayed.Hom[ simple-ct ct ]-set {Γ} {Δ} u X Y = Hom-set (Γ ⊗ X .fst) (Y .fst)
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/CT-Structure.lagda.md:57]:
```suggestion
simple-ct ct .Displayed.Ob[_] Γ = Σ[ X ∈ Ob ] ∣ ct .is-tp X ∣
simple-ct ct .Displayed.Hom[_] {Γ} {Δ} u X Y = Hom (Γ ⊗ X .fst) (Y .fst)
 simple-ct ct .Displayed.Hom[_]-set {Γ} {Δ} u X Y = Hom-set (Γ ⊗ X .fst) (Y .fst)
```

File: src/Cat/Displayed/Instances/CT-Structure.lagda.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+simple-ct ct .Displayed.id′ = π₂
+simple-ct ct .Displayed._∘′_ {f = u} {g = v} f g = f ∘ ⟨ v ∘ π₁ , g ⟩ 
+simple-ct ct .Displayed.idr′ {f = u} f =
+  f ∘ ⟨ (id ∘ π₁) , π₂ ⟩ ≡⟨ products! B has-prods ⟩
+  f                      ∎
+simple-ct ct .Displayed.idl′ {f = u} f =
+  π₂ ∘ ⟨ u ∘ π₁ , f ⟩ ≡⟨ products! B has-prods ⟩
+  f                   ∎
+simple-ct ct .Displayed.assoc′ {f = u} {g = v} {h = w} f g h =
+  f ∘ ⟨ (v ∘ w) ∘ π₁ , g ∘ ⟨ w ∘ π₁ , h ⟩ ⟩ ≡⟨ products! B has-prods ⟩
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/CT-Structure.lagda.md:67]: hell yeah

File: src/Cat/Displayed/Instances/Simple.lagda.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+When constructing models of type theories, the general pattern
+is to construct a category of contexts and substitutions, and then
+to study types and terms as structures over this category.
+The language of displayed categories allows us to capture this situation
+quite succinctly by considering these extra pieces of equippment as
+being fibred over our category of contexts.
+
+Focusing in, the language of *simple fibrations* provides us with enough
+structure to study simply-typed languages that have enough structure
+to represent contexts internally (IE: product types).
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/Simple.lagda.md:31]:
```suggestion
to represent contexts internally (i.e.: product types).
```

File: src/Cat/Displayed/Instances/Simple.lagda.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+To start, we fix some base category $\ca{B}$ with binary products.
+Intuitvely, this will be some sort of category of contexts, with the
+product structure given by context extension. We can then consider
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/Simple.lagda.md:35]:
```suggestion
products interpreting context extension. We can then consider
```

File: src/Cat/Displayed/Instances/Simple.lagda.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+a type in a context to be an object $\Gamma \times X : \ca{B}$.
+Maps between types in contexts $(\Gamma \times X) \to \Delta \times Y$
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/Simple.lagda.md:37]:
```suggestion
Maps between types in contexts $(\Gamma \times X) \to (\Delta \times Y)$
```

File: src/Cat/Displayed/Instances/Simple.lagda.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+by playing type-tetris, as it's all a matter of golfing the types
+and contexts into the correct place. The category laws are then a matter
+of bashing through a bunch of nested pairings and projections, and
+can be entirely automated.
+
+```agda
+simple : Displayed B o ℓ
+Displayed.Ob[ simple ] Γ = Ob
+Displayed.Hom[ simple ] {Γ} {Δ} u X Y = Hom (Γ ⊗ X) Y
+Displayed.Hom[ simple ]-set _ _ _ = Hom-set (_ ⊗ _) _
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/Simple.lagda.md:68]:
```suggestion
simple .Displayed.Ob[_] Γ = Ob
simple .Displayed.Hom[_] {Γ} {Δ} u X Y = Hom (Γ ⊗ X) Y
simple .Displayed.Hom[_]-set _ _ _ = Hom-set (_ ⊗ _) _
```

File: src/Cat/Displayed/Instances/Simple.lagda.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+# Fibration Structure
+
+As suggested by it's name, the simple fibration is a fibration.
+given a map $\Gamma \to Delta$ in the base, and a $(\Delta , Y)$
+upstairs, we can construct a lift by selecting $(\Gamma, Y)$ as the
+corner of the lift, and then using the second projection as the lift
+itself. Intuitively, this encodes a substitution of contexts: because
+we are working with a simple type theory, the substitutions don't need
+to touch the types, as there is no possible dependencies!
```
Comment (plt-amy) [inline src/Cat/Displayed/Instances/Simple.lagda.md:90]:
```suggestion
to touch the types, as there are no possible dependencies!
```

File: src/Cat/Displayed/Instances/CT-Structure.lagda.md (hunk 3 above)
Comment (TOTBWF) [reply src/Cat/Displayed/Instances/CT-Structure.lagda.md:67]: 😎

**Pull request merged**